In [1]:

import sys
import os
sys.path.append(os.path.abspath('..'))
from tools.component_tools import GLC,GLC_Gas,Fluid
import numpy as np 
R_const=8.314
T = [673, 673, 673, 673, 713, 713, 713, 713, 713]
Z = 0.6
R = 0.0547/2
Q_l = [
    71.0    * 1e-3 / 3600,
    66      * 1e-3 / 3600,
    69      * 1e-3 / 3600,
    100     * 1e-3 / 3600,
    61      * 1e-3 / 3600,
    68      * 1e-3 / 3600,
    68      * 1e-3 / 3600,
    100     * 1e-3 / 3600,
    70      * 1e-3 / 3600,
]
Q_gas = [
    3 * 1e-3 / 3600,
    6 * 1e-3 / 3600,
    9 * 1e-3 / 3600,
    6 * 1e-3 / 3600,
    3 * 1e-3 / 3600,
    6 * 1e-3 / 3600,
    9 * 1e-3 / 3600,
    6 * 1e-3 / 3600,
    6 * 1e-3 / 3600,
]
p_in = [1085, 1003, 927, 1188, 1258, 1152, 1158, 1228, 2490]
p_out = [653, 613, 557, 858, 682, 670, 612, 712, 1215]
B_vec = []
kl_vec = []
for i in range(len(Q_gas)):
    SweepGas=GLC_Gas(G_gas=Q_gas[i],pg_in=0,p_tot=1.5E5)
    
    PbLi=Fluid(Solubility=1.33E-4*np.exp(-1350/R_const/T[i]),MS=False )
    c_in=p_in[i]**0.5*PbLi.Solubility
    Melodie=GLC(H=Z,R=R,c_in=p_in[i]**0.5*PbLi.Solubility,fluid=PbLi,GLC_gas=SweepGas,T=T[i],G_L=Q_l[i],c_out=p_out[i]**0.5*PbLi.Solubility)
    [a,b] = Melodie.get_kla_from_cout()
    print(a, b)
    B_vec.append(a)
    kl_vec.append(b)



30.212997495461888 0.004873070573395917
28.085321615499783 0.003928596589236866
29.361927143477047 0.00412525021157648
42.5535175992421 0.0038782808763153454
25.957645735537678 0.005054425415139703
28.936391967484624 0.004499560386868871
28.936391967484624 0.00516701987252399
42.5535175992421 0.006985647996240323
29.787462319469466 0.006115638885991319


if done correctly the results should be the same as this table
| B         | kla$$10^3$$    |
|-----------|----------------|
| 30.2      | 4.87           |
| 28.0      | 3.92           |
| 29.4      | 4.13           |
| 42.6      | 3.88           |
| 26.0      | 5.06           |
| 28.9      | 4.50           |
| 28.9      | 5.16           |
| 42.6      | 6.99           |
| 29.8      | 6.12           |
| 28.2      | 2.84           |
| 14.1      | 2.12           |
| 14.1      | 1.90           |
| 14.1      | 1.66           |
| 14.1      | 1.73           |



In [2]:
# reset notebook
%reset -f
from tools.component_tools import GLC,GLC_Gas,Fluid
import numpy as np 
R_const=8.31
T = [673, 673, 673, 673, 673]
Z = 0.8
R = 0.03
Q_l = [
    80.0 * 1e-3 / 3600,
    40   * 1e-3 / 3600,
    40   * 1e-3 / 3600,
    40   * 1e-3 / 3600,
    40   * 1e-3 / 3600,
]
Q_gas = [
    6 * 1e-3 / 3600,
    6 * 1e-3 / 3600,
    30 * 1e-3 / 3600,
    6 * 1e-3 / 3600,
    6 * 1e-3 / 3600,
]
p_in = [1275, 1050, 988, 463, 225]
p_out = [796, 515, 484, 267, 130]
B_vec = []
kl_vec = []

for i in range(len(Q_gas)):
    SweepGas=GLC_Gas(G_gas=Q_gas[i],pg_in=0,p_tot=1.5E5)
    
    PbLi=Fluid(Solubility=1.33E-4*np.exp(-1350/R_const/T[i]),MS=False )
    c_in=p_in[i]**0.5*PbLi.Solubility
    Melodie=GLC(H=Z,R=R,c_in=p_in[i]**0.5*PbLi.Solubility,fluid=PbLi,GLC_gas=SweepGas,T=T[i],G_L=Q_l[i],c_out=p_out[i]**0.5*PbLi.Solubility)
    [a,b] = Melodie.get_kla_from_cout()
    print(a, b)
    B_vec.append(a)
    kl_vec.append(b)


28.294212105225842 0.002847199560353614
14.147106052612921 0.002126282086634865
14.147106052612921 0.0019059006958863619
14.147106052612921 0.0016619703471135925
14.147106052612921 0.0017339951739678446
